km/km_signal.c

```diff
@@ -1067,6 +1089,9 @@ int km_sig_snapshot_recover(char* buf, size_t length)
       km_warnx("km_nt_sighand_t size mismatch - old snapshot?");
       return -1;
    }
+   if (nt_sa->signo == NSIG) {
```

Comment:
I think you want to compare against KM_SIGSNAPCREATE here.

Reply:
yep